src/utilities/buildASTSchema.js

```diff
@@ -254,12 +229,12 @@ export function buildASTSchema(ast: DocumentNode): GraphQLSchema {
   }
 
   return new GraphQLSchema({
-    query: getObjectType(nodeMap[queryTypeName]),
-    mutation: mutationTypeName ?
-      getObjectType(nodeMap[mutationTypeName]) :
+    query: getObjectType(operationTypes['query']),
+    mutation: operationTypes['mutation'] ?
+      getObjectType(operationTypes['mutation']) :
       null,
-    subscription: subscriptionTypeName ?
-      getObjectType(nodeMap[subscriptionTypeName]) :
+    subscription: operationTypes['subscription'] ?
+      getObjectType(operationTypes['subscription']) :
```

Comment:
Same as above. Am I misunderstanding something? It looks like we aren't being consistent with the operation names (query vs. Query and etc)

Reply:
@wincent Fully remove strings now it: `operationTypes.subscription`.

> consistent with the operation names (query vs. Query and etc)

It's how spec working, inside schema definition it's `query` operation but if schema definition is absent then you should look for `Query` type. So naming is following the specification.